File: Lecture12-Gradient/main.cpp
#include <opencv4/opencv2/opencv.hpp>

using namespace cv;

void gradient()
{
    Mat img = imread("/home/berkan/CLionProjects/OpenCV/Ders12_Gradient/word.jpg", IMREAD_GRAYSCALE);
    Mat filter, erodeImg, dilateImg, gradientImg;
    namedWindow("Gradient");
    threshold(img, img, 128, 255, THRESH_BINARY);
    filter = Mat::ones(Size(3, 3), img.type());

    int min = 0;
    createTrackbar("gradient", "Gradient", &min, 10);

    while (true)
    {
        erode(img, erodeImg, filter, Point(-1, -1), min, 0, Scalar(255, 255, 255));
        dilate(img, dilateImg, filter, Point(-1, -1), min, 0, Scalar(255, 255, 255));
        gradientImg = dilateImg - erodeImg;
        imshow("Gradient", gradientImg);

        if (waitKey(10) == 27)
        {
            break;
        }
    }
    destroyWindow("Gradient");
}

int main() {
    gradient();
    return 0;
}

File: Lecture12-Gradient/main.py
import cv2
import numpy as np

def nothing():
    pass

def gradient():
    img = cv2.imread("cumle.jpg", cv2.IMREAD_GRAYSCALE)
    cv2.namedWindow("Gradient", cv2.WINDOW_AUTOSIZE)
    cv2.threshold(img, 128, 255, cv2.THRESH_BINARY)
    cv2.createTrackbar("Gradient", "Gradient", 0, 15, nothing)
    filtre = np.ones((3,3))

    while True:
        iteration = cv2.getTrackbarPos("Gradient", "Gradient")

        dil = cv2.dilate(img, filtre, dst = (-1,-1), iterations=iteration)
        ero = cv2.erode(img, filtre, dst = (-1,-1), iterations=iteration)

        gradient = dil - ero

        cv2.imshow("Gradient", gradient)

        if cv2.waitKey(10) == 27:
            break

    cv2.destroyAllWindows()


if __name__ == '__main__':
    gradient()